common/flatpak-installation.c

```diff
@@ -1190,6 +1196,29 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
                                                                             full_ref,
                                                                             cancellable))
             g_ptr_array_add (updates, g_object_ref (installed_ref));
+
+          /* Check for runtime: Now see if any of the installed ref is missing its runtime component.
+           * If so, return that installed ref in the updates list, so that FlatpakTransaction
+           * can resolve one of its operation to install the runtime instead.
+           *
+           * Runtime of an app can go missing if an app upgrade makes an app dependent on a new runtime
+           * entirely. We had couple of cases like that in the past, for example, before it was updated
+           * to use FlatpakTransaction, updating an app in GNOME Software to a version which needs a
+           * different runtime would not install that new runtime, leaving the app unusable.
+           */
+          deploy_data = flatpak_dir_get_deploy_data (dir, full_ref, FLATPAK_DEPLOY_VERSION_CURRENT, cancellable, &local_error);
+          if (local_error != NULL)
```

Comment:
In these kind of fixups i don't think the errors and whatnot are useful and just complicate the code. I'd recommend not passing local_error and just move all the code below into a `if (deploy_data != NULL) { ... }`